test/storage/index_manager_test.cpp

```diff
@@ -0,0 +1,117 @@
+  IndexManager *index_manager_;
+};
+
+// Check the correctness of drop index
+// NOLINTNEXTLINE
+TEST_F(IndexManagerTest, DropIndexCorrectnessTest) {
+  auto txn0 = txn_manager_->BeginTransaction();
+  auto catalog_ = new catalog::Catalog(txn_manager_, txn0);
+
+  // terrier has db_oid_t DEFAULT_DATABASE_OID
+  const catalog::db_oid_t terrier_oid(catalog::DEFAULT_DATABASE_OID);
+  auto db_handle = catalog_->GetDatabaseHandle();
+  auto ns_handle = db_handle.GetNamespaceHandle(txn0, terrier_oid);
+  auto table_handle = ns_handle.GetTableHandle(txn0, "public");
+  auto ns_oid = ns_handle.NameToOid(txn0, std::string("public"));
+
+  // define schema
+  std::vector<catalog::Schema::Column> cols;
+  cols.emplace_back("sex", type::TypeId::BOOLEAN, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("id", type::TypeId::INTEGER, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("name", type::TypeId::VARCHAR, 100, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("address", type::TypeId::VARCHAR, 200, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  catalog::Schema schema(cols);
+
+  // create table
+  auto table = table_handle.CreateTable(txn0, schema, "test_table");
+  auto table_oid = table_handle.NameToOid(txn0, "test_table");
+  auto table_entry = table_handle.GetTableEntry(txn0, "test_table");
+  EXPECT_NE(table_entry, nullptr);
+  std::string_view str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(0));
+  EXPECT_EQ(str, "public");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(1));
+  EXPECT_EQ(str, "test_table");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(2));
+  EXPECT_EQ(str, "pg_default");
+
+  // Insert a few rows into the table
+  auto ptr = table_handle.GetTable(txn0, "test_table");
+  EXPECT_EQ(ptr, table);
+
+  for (int i = 0; i < 200; ++i) {
+    std::vector<type::TransientValue> row;
+    std::ostringstream stringStream;
+    stringStream << "name_" << i;
+    std::string name_string = stringStream.str();
+    stringStream << "address_" << i;
+    std::string address_string = stringStream.str();
+    row.emplace_back(type::TransientValueFactory::GetBoolean(i % 2 == 0));
+    row.emplace_back(type::TransientValueFactory::GetInteger(i));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(name_string));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(address_string));
+    ptr->InsertRow(txn0, row);
+  }
+
+  // Commit the setting transaction
+  txn_manager_->Commit(txn0, TestCallbacks::EmptyCallback, nullptr);
+
+  // Set up index attributes and key attributes
+  std::vector<std::string> index_attrs{"sex", "id", "name"};
+  std::vector<std::string> key_attrs{"id", "name"};
+
+  // Create the index
+  auto index_oid = index_manager_->CreateConcurrently(terrier_oid, ns_oid, table_oid, parser::IndexType::BWTREE, false,
```

Comment:
When you are calling `CreateConcurrently`, there is no active txns running, so you didn't actually test whether `CreateConcurrently` successfully build an index on its visible version. At least, I would suggest running `CreateConcurrently` when there are some active txns.